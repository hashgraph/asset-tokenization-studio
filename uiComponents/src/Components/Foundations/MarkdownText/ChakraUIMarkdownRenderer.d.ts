import type { Components } from "react-markdown/src/ast-to-react";
export interface Defaults extends Components {
    heading?: Components["h1"];
}
export declare const defaults: Defaults;
export declare function ChakraUIMarkdownRenderer(theme?: Defaults, merge?: boolean): Components;
