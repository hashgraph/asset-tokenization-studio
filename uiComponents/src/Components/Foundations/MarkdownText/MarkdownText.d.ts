import type { BoxProps } from "@chakra-ui/layout";
import type { ComponentWithAs } from "@chakra-ui/system";
import React from "react";
import type { TransformLinkTarget } from "react-markdown/src/ast-to-react";
import type { Defaults } from "./ChakraUIMarkdownRenderer";
export interface MarkdownTextProps {
    children: string;
    styles?: BoxProps;
    sx?: BoxProps["sx"];
    linkTarget?: TransformLinkTarget | React.HTMLAttributeAnchorTarget;
    theme?: Defaults;
}
export declare const MarkdownText: ComponentWithAs<"div", MarkdownTextProps>;
