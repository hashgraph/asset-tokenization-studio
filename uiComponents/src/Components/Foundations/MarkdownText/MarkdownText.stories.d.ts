import React from "react";
declare const meta: import("@storybook/types").ComponentAnnotations<import("@storybook/react/dist/types-0a347bb9").R, import("@chakra-ui/system/dist/system.types").MergeWithAs<React.DetailedHTMLProps<React.HTMLAttributes<HTMLDivElement>, HTMLDivElement>, any, import("./MarkdownText").MarkdownTextProps, import("@chakra-ui/system/dist/system.types").As>>;
export default meta;
export declare const Default: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0a347bb9").R, import("@chakra-ui/system/dist/system.types").MergeWithAs<React.DetailedHTMLProps<React.HTMLAttributes<HTMLDivElement>, HTMLDivElement>, any, import("./MarkdownText").MarkdownTextProps, import("@chakra-ui/system/dist/system.types").As>>;
export declare const WithCustomTheme: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0a347bb9").R, import("@chakra-ui/system/dist/system.types").MergeWithAs<React.DetailedHTMLProps<React.HTMLAttributes<HTMLDivElement>, HTMLDivElement>, any, import("./MarkdownText").MarkdownTextProps, import("@chakra-ui/system/dist/system.types").As>>;
