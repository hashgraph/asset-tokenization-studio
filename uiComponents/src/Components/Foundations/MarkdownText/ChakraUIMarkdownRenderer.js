var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
import { Checkbox } from "@chakra-ui/checkbox";
import { Image } from "@chakra-ui/image";
import { Code, Divider, Link, ListItem, OrderedList, UnorderedList, } from "@chakra-ui/layout";
import { chakra } from "@chakra-ui/system";
import { Table, Tbody, Td, Th, Thead, Tr } from "@chakra-ui/table";
import _merge from "lodash/merge";
import * as React from "react";
import { Heading } from "../Heading";
import { Text } from "../Text";
function getCoreProps(props) {
    return props["data-sourcepos"]
        ? { "data-sourcepos": props["data-sourcepos"] }
        : {};
}
export var defaults = {
    p: function (props) {
        var children = props.children;
        return React.createElement(Text, { mb: 2 }, children);
    },
    em: function (props) {
        var children = props.children;
        return React.createElement(Text, { as: "em" }, children);
    },
    blockquote: function (props) {
        var children = props.children;
        return (React.createElement(Code, { as: "blockquote", p: 2 }, children));
    },
    code: function (props) {
        var inline = props.inline, children = props.children, className = props.className;
        if (inline) {
            return React.createElement(Code, { p: 2, children: children });
        }
        return (React.createElement(Code, { className: className, whiteSpace: "break-spaces", display: "block", w: "full", p: 2, children: children }));
    },
    del: function (props) {
        var children = props.children;
        return React.createElement(Text, { as: "del" }, children);
    },
    hr: function () {
        return React.createElement(Divider, null);
    },
    a: Link,
    img: Image,
    text: function (props) {
        var children = props.children;
        return React.createElement(Text, { as: "span" }, children);
    },
    ul: function (props) {
        var ordered = props.ordered, children = props.children, depth = props.depth;
        var attrs = getCoreProps(props);
        var Element = UnorderedList;
        var styleType = "disc";
        if (ordered) {
            Element = OrderedList;
            styleType = "decimal";
        }
        if (depth === 1)
            styleType = "circle";
        return (React.createElement(Element, __assign({ spacing: 2, as: ordered ? "ol" : "ul", styleType: styleType, pl: 4 }, attrs), children));
    },
    ol: function (props) {
        var ordered = props.ordered, children = props.children, depth = props.depth;
        var attrs = getCoreProps(props);
        var Element = UnorderedList;
        var styleType = "disc";
        if (ordered) {
            Element = OrderedList;
            styleType = "decimal";
        }
        if (depth === 1)
            styleType = "circle";
        return (React.createElement(Element, __assign({ spacing: 2, as: ordered ? "ol" : "ul", styleType: styleType, pl: 4 }, attrs), children));
    },
    li: function (props) {
        var children = props.children, checked = props.checked;
        var checkbox = null;
        if (checked !== null && checked !== undefined) {
            checkbox = (React.createElement(Checkbox, { isChecked: checked, isReadOnly: true }, children));
        }
        return (React.createElement(ListItem, __assign({}, getCoreProps(props), { listStyleType: checked !== null ? "none" : "inherit" }), checkbox || children));
    },
    heading: function (props) {
        var level = props.level, children = props.children;
        var sizes = ["2xl", "xl", "lg", "md", "sm", "xs"];
        return (React.createElement(Heading, __assign({ my: 4, 
            // @ts-ignore
            as: "h".concat(level), size: sizes[level - 1] }, getCoreProps(props)), children));
    },
    pre: function (props) {
        var children = props.children;
        return React.createElement(chakra.pre, __assign({}, getCoreProps(props)), children);
    },
    table: Table,
    thead: Thead,
    tbody: Tbody,
    tr: function (props) { return React.createElement(Tr, null, props.children); },
    td: function (props) { return React.createElement(Td, null, props.children); },
    th: function (props) { return React.createElement(Th, null, props.children); }
};
export function ChakraUIMarkdownRenderer(theme, merge) {
    if (merge === void 0) { merge = true; }
    var elements = {
        p: defaults.p,
        em: defaults.em,
        blockquote: defaults.blockquote,
        code: defaults.code,
        del: defaults.del,
        hr: defaults.hr,
        a: defaults.a,
        img: defaults.img,
        text: defaults.text,
        ul: defaults.ul,
        ol: defaults.ol,
        li: defaults.li,
        h1: defaults.heading,
        h2: defaults.heading,
        h3: defaults.heading,
        h4: defaults.heading,
        h5: defaults.heading,
        h6: defaults.heading,
        pre: defaults.pre,
        table: defaults.table,
        thead: defaults.thead,
        tbody: defaults.tbody,
        tr: defaults.tr,
        td: defaults.td,
        th: defaults.th
    };
    if (theme && merge) {
        return _merge(elements, theme);
    }
    return elements;
}
