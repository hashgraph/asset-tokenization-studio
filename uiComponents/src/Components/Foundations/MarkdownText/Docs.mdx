import {
  Story,
  Meta,
  Title,
  Controls,
  Primary,
  Stories,
  ArgTypes,
} from "@storybook/addon-docs";
import { Grid, GridItem, Text, Link, Center, Flex } from "@chakra-ui/react";
import * as MarkdownTextStories from "./MarkdownText.stories";

<Meta of={MarkdownTextStories} />

<Title />
<Primary />

## Props

<ArgTypes of={MarkdownTextStories} />

## Stories

<Stories title="" />
