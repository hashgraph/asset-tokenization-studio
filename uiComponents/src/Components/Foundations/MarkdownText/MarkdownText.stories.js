var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
import React from "react";
import { Heading } from "../Heading";
import { MarkdownText } from "./MarkdownText";
var meta = {
    title: "Design System/Foundations/MarkdownText",
    component: MarkdownText,
    argTypes: {},
    parameters: {
        docs: {}
    },
    args: {}
};
export default meta;
var Template = function (args) {
    return React.createElement(MarkdownText, __assign({}, args));
};
export var Default = Template.bind({});
Default.args = {
    children: "\n  #### H4 Heading\n  #### _H4 Heading Italic_\n  "
};
export var WithCustomTheme = Template.bind({});
WithCustomTheme.args = {
    children: "# h1 Heading 8-) \n  ## h2 Heading\n  ### h3 Heading\n  #### h4 Heading\n  ##### h5 Heading\n  ###### h6 Heading\n  ",
    theme: {
        h2: function (props) { return React.createElement(Heading, { color: "red" }, props.children); }
    }
};
