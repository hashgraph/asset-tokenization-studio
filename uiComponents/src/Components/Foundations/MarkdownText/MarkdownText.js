var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
import { Box } from "@chakra-ui/layout";
import { forwardRef } from "@chakra-ui/system";
import React from "react";
import ReactMarkdown from "react-markdown";
import { ChakraUIMarkdownRenderer } from "./ChakraUIMarkdownRenderer";
export var MarkdownText = forwardRef(function (_a, ref) {
    var styles = _a.styles, children = _a.children, sx = _a.sx, linkTarget = _a.linkTarget, theme = _a.theme;
    return (React.createElement(Box, __assign({ ref: ref }, styles, { sx: sx }),
        React.createElement(ReactMarkdown, { linkTarget: linkTarget, components: ChakraUIMarkdownRenderer(theme) }, children)));
});
