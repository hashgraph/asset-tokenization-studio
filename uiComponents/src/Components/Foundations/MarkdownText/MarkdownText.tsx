import type { BoxProps } from "@chakra-ui/layout";
import type { ComponentWithAs } from "@chakra-ui/system";
import { Box } from "@chakra-ui/layout";
import { forwardRef } from "@chakra-ui/system";
import React from "react";
import ReactMarkdown from "react-markdown";
import type { TransformLinkTarget } from "react-markdown/src/ast-to-react";
import type { Defaults } from "./ChakraUIMarkdownRenderer";
import { ChakraUIMarkdownRenderer } from "./ChakraUIMarkdownRenderer";

export interface MarkdownTextProps {
  children: string;
  styles?: BoxProps;
  sx?: BoxProps["sx"];
  linkTarget?: TransformLinkTarget | React.HTMLAttributeAnchorTarget;
  theme?: Defaults;
}

export const MarkdownText: ComponentWithAs<"div", MarkdownTextProps> =
  forwardRef<MarkdownTextProps, "div">(
    ({ styles, children, sx, linkTarget, theme }: MarkdownTextProps, ref) => {
      return (
        <Box ref={ref} {...styles} sx={sx}>
          <ReactMarkdown
            linkTarget={linkTarget}
            components={ChakraUIMarkdownRenderer(theme)}
          >
            {children}
          </ReactMarkdown>
        </Box>
      );
    }
  );
