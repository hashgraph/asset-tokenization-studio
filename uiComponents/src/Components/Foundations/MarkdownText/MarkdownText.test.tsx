import { render } from "@/test-utils";
import React from "react";
import { Text } from "../Text";
import type { MarkdownTextProps } from "./MarkdownText";
import { MarkdownText } from "./MarkdownText";

describe(`<MarkdownText />`, () => {
  const children = `# h1 Heading 8-) 
  ## h2 Heading
  ### h3 Heading
  #### h4 Heading
  ##### h5 Heading
  ###### h6 Heading
  Simple paragraph
  `;

  const defaultProps: MarkdownTextProps = {
    children,
    theme: {
      p: (props) => <Text color="blue">{props.children}</Text>,
    },
  };

  const factoryComponent = (props: MarkdownTextProps) =>
    render(<MarkdownText {...props} />);

  test("renders correctly", () => {
    const component = factoryComponent({ ...defaultProps });
    expect(component.asFragment()).toMatchSnapshot();
  });
});
