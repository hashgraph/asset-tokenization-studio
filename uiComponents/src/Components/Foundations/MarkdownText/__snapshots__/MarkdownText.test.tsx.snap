// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MarkdownText /> renders correctly 1`] = `
<DocumentFragment>
  <div
    class="css-0"
  >
    <h1
      class="css-1ftzwd4"
    >
      h1 Heading 8-)
    </h1>
    

    <h2
      class="css-6rvmc6"
    >
      h2 Heading
    </h2>
    

    <h3
      class="css-127uure"
    >
      h3 Heading
    </h3>
    

    <h4
      class="css-k61gls"
    >
      h4 Heading
    </h4>
    

    <h5
      class="css-1rc0ucf"
    >
      h5 Heading
    </h5>
    

    <h6
      class="css-xgxc3b"
    >
      h6 Heading
    </h6>
    

    <p
      class="css-14ksm7b"
    >
      Simple paragraph
    </p>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;
